app/controllers/providers/application_merits_task/has_other_opponents_controller.rb

```diff
@@ -2,35 +2,57 @@ module Providers
   module ApplicationMeritsTask
     class HasOtherOpponentsController < ProviderBaseController
       def show
-        form
+        return redirect_to providers_legal_aid_application_opponent_type_path(legal_aid_application) unless opponents.any?
+
+        @form = LegalAidApplications::HasOtherOpponentsForm.new(model: legal_aid_application)
       end
 
       def update
+        @form = LegalAidApplications::HasOtherOpponentsForm.new(form_params)
         update_task(:application, :opponent_name)
         return continue_or_draft if draft_selected?
-        return go_forward(form.has_other_opponents?) if form.valid?
+        return go_forward(@form.has_other_opponents?) if @form.valid?
+
+        render :show
+      end
+
+      def destroy
+        opponent&.destroy!
```

Comment:
is the safe operator needed? if there was no opponent you would probably want to stop/error and do something else in any event?!

Reply:
No I don't believe it is needed. Just over caution. Have removed 👍